ql/currency.hpp

```diff
@@ -81,6 +83,8 @@ namespace QuantLib {
         bool empty() const;
         //! currency used for triangulated exchange when required
         const Currency& triangulationCurrency() const;
+        //! minor unit codes, e.g. GBp, GBX for GBP
+        const std::set<std::string>& minorUnitCodes() const;
```

Comment:
Is it valid for the set to be empty? If not, then how about making it `boost::optional<std::set<std::string>>` and asserting the set is not empty when it has a value?

Reply:
Hm. What do we really gain?

Reply:
Mostly just clarity on whether it's valid for the set to be empty or not, but I suppose a similar result can be achieved with more source code comments.

For example, is the currency code itself (e.g. GBP) always implicitly considered a minor unit code? What if it's explicitly included in the set? And who is responsible for populating the set - the currency definition in QuantLib, the library user, or both?

Reply:
The set can be empty, e.g. there are no minor units for XAU.

The currency code is not itself considered a minor unit code, including it in the set would be wrong. In general "fractionsPerUnit" minor units make one unit of the currency.

The responsibility to populate the set is the currency definition, this is not different from the other data. It's not part of this merge request to provide this data in all derived currencies in QuantLib though, here we only extend the base class to be able to store the code.

Reply:
Ok!

Reply:
If we want to make this change, we should probably open another ticket to go through the existing currencies and add the minor unit codes to them everywhere - a nice "help wanted" and "good first issue" - ticket probably